### PR TITLE
🌱  Avoid cd'ing into test folders when building Docker images

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -801,13 +801,13 @@ docker-build-kubeadm-control-plane: ## Build the docker image for kubeadm contro
 
 .PHONY: docker-build-docker-infrastructure
 docker-build-docker-infrastructure: ## Build the docker image for docker infrastructure controller manager
-	cd $(CAPD_DIR); DOCKER_BUILDKIT=1 docker build --build-arg builder_image=$(GO_CONTAINER_IMAGE) --build-arg goproxy=$(GOPROXY) --build-arg ARCH=$(ARCH) --build-arg ldflags="$(LDFLAGS)" ../../.. -t $(CAPD_CONTROLLER_IMG)-$(ARCH):$(TAG) --file Dockerfile
+	DOCKER_BUILDKIT=1 docker build --build-arg builder_image=$(GO_CONTAINER_IMAGE) --build-arg goproxy=$(GOPROXY) --build-arg ARCH=$(ARCH) --build-arg ldflags="$(LDFLAGS)" . -t $(CAPD_CONTROLLER_IMG)-$(ARCH):$(TAG) --file ./test/infrastructure/docker/Dockerfile
```
Comment (fabriziopandini) [inline Makefile:804]: q: is it correct to keep using . as context (we are basically using loading the entire project in the build context)

**Review by chrischdi — COMMENTED** [review]
/lgtm

File: Makefile [hunk 2]
```diff
@@ -801,13 +801,13 @@ docker-build-kubeadm-control-plane: ## Build the docker image for kubeadm contro
...
 	$(MAKE) set-manifest-image MANIFEST_IMG=$(CAPD_CONTROLLER_IMG)-$(ARCH) MANIFEST_TAG=$(TAG) TARGET_RESOURCE="$(CAPD_DIR)/config/default/manager_image_patch.yaml"
 	$(MAKE) set-manifest-pull-policy TARGET_RESOURCE="$(CAPD_DIR)/config/default/manager_pull_policy.yaml"
 
 .PHONY: docker-build-in-memory-infrastructure
 docker-build-in-memory-infrastructure: ## Build the docker image for in-memory infrastructure controller manager
-	cd $(CAPIM_DIR); DOCKER_BUILDKIT=1 docker build --build-arg builder_image=$(GO_CONTAINER_IMAGE) --build-arg goproxy=$(GOPROXY) --build-arg ARCH=$(ARCH) --build-arg ldflags="$(LDFLAGS)" ../../.. -t $(CAPIM_CONTROLLER_IMG)-$(ARCH):$(TAG) --file Dockerfile
+	DOCKER_BUILDKIT=1 docker build --build-arg builder_image=$(GO_CONTAINER_IMAGE) --build-arg goproxy=$(GOPROXY) --build-arg ARCH=$(ARCH) --build-arg ldflags="$(LDFLAGS)" . -t $(CAPIM_CONTROLLER_IMG)-$(ARCH):$(TAG) --file ./test/infrastructure/inmemory/Dockerfile
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply Makefile]: Thx, done

File: Makefile (hunk 1 above)
Comment (sbueringer) [reply Makefile]: We just moved to . on this PR

**Comment by sbueringer** [comment]
Pushed a fix

File: Makefile (hunk 1 above)
Comment (fabriziopandini) [reply Makefile]: it is, I just saw that before the context was ../../.., so it was already the entire project

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (nawazkh) [reply Makefile]: But I see why you could have hard-coded `./test/infrastructure/inmemory/Dockerfile`. Probably to check if the make target works fine. 
You can ignore my above suggestion, not a major one.

**Pull request merged**